packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -485,3 +488,85 @@ export const WithLabelInternally = () => {
     />
   )
 }
+
+export const AsyncFetch: StoryObj<SelectPanelPropsAndCustomArgs> = {
+  render: ({initialLoadingType, height, componentManagesLoading}: SelectPanelPropsAndCustomArgs) => {
+    const [loading, setLoading] = React.useState<boolean>(true)
+    const [selected, setSelected] = React.useState<ItemInput[]>([])
+    const [filteredItems, setFilteredItems] = React.useState<ItemInput[]>([])
+    const [open, setOpen] = useState(false)
+    const filterTimerId = useRef<number | null>(null)
+    const {safeSetTimeout, safeClearTimeout} = useSafeTimeout()
+
+    const fetchItems = (query: string) => {
+      setLoading(true)
+
+      if (filterTimerId.current) {
+        safeClearTimeout(filterTimerId.current)
+      }
+
+      filterTimerId.current = safeSetTimeout(() => {
+        setFilteredItems(items.filter(item => item.text.toLowerCase().startsWith(query.toLowerCase())))
+        setLoading(false)
+      }, 2000) as unknown as number
+    }
+
+    const onOpenChange = (value: boolean) => {
+      setLoading(true)
+      setOpen(value)
+      fetchItems('')
+    }
+
+    const loadingProps = {
+      initialLoadingType,
+      ...(componentManagesLoading ? {} : {loading}),
+    }
+
+    return (
+      <SelectPanel
+        title="Select labels"
+        subtitle="Use labels to organize issues and pull requests"
+        renderAnchor={({children, 'aria-labelledby': ariaLabelledBy, ...anchorProps}) => (
+          <Button
+            trailingAction={TriangleDownIcon}
+            aria-labelledby={` ${ariaLabelledBy}`}
+            {...anchorProps}
+            aria-haspopup="dialog"
+          >
+            {children ?? 'Select Labels'}
+          </Button>
+        )}
+        placeholderText="Filter labels"
+        open={open}
+        onOpenChange={onOpenChange}
+        items={filteredItems}
+        selected={selected}
+        onSelectedChange={setSelected}
+        onFilterChange={fetchItems}
+        showItemDividers={true}
+        height={height}
+        {...loadingProps}
+      />
+    )
+  },
+  args: {
+    initialLoadingType: 'spinner',
+    height: 'medium',
+    componentManagesLoading: true,
+  },
+  argTypes: {
+    initialLoadingType: {
+      control: 'select',
+      options: ['spinner', 'skeleton'],
+    },
+    height: {
+      control: 'select',
+      options: ['auto', 'xsmall', 'small', 'medium', 'large', 'xlarge'],
+    },
+    componentManagesLoading: {
```

Comment:
would adding a control for 'loading' here as well make the difference between the two evident? it's really hard to grasp the difference between the two off this control in the UI

Reply:
I wonder if perhaps we don't need this option... I added it for testing purposes, but I can't really see it being useful to us in the future, or to outside contributors.